pytorch_lightning/trainer/states.py

```diff
@@ -0,0 +1,31 @@
+from enum import Enum, auto
+
+from pytorch_lightning import Callback
+
+
+class TrainerState(Enum):
+    """ State which is set to the Trainer to indicate what is being executed. """
+    INITIALIZE = auto()
+    RUNNING = auto()
+    FINISHED = auto()
+
+
+class _TrainerStateSwitcher(Callback):
```

Comment:
I would rather "hardcode" to the trainer so no one will accidentally remove this callback...

Reply:
Done